### PR TITLE
Add pagination support to async Select

### PR DESCRIPTION
**Pull request by Etheryte**
## What does this PR change?

First part of addressing https://github.com/SUSE/spacewalk/issues/16363, this PR adds pagination to the async dropdown and then switches the channel selection over from the legacy select logic so it doesn't try to render 5000 items at once. I haven't tested the new addition with all possible combinations of props that the dropdown component accepts, but for the given case it seems to work fine. In short, I expect there might be some edge case bugs that I haven't found yet, but this is the first step to address the performance problems of this UI.  

In the future, the logic for the pagination can be moved to the server, but currently I don't think it's worth putting in that effort yet until we know what other changes we might need.  

No changelog yet since this doesn't fix the whole problem.

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [x] **DONE**

## Test coverage
- No tests: This definitely needs some tests, but I'll add some in a followup PR once I've figured out the big parts of fixing this L3.

- [x] **DONE**

## Links

Part of https://github.com/SUSE/spacewalk/issues/16363

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: web/html/src/manager/content-management/shared/components/panels/sources/channels/child-channels.tsx [hunk 1]
```diff
@@ -24,47 +24,48 @@ const ChildChannels = (props: PropsType) => {
...
-              {isMandatory ? (
-                <span className="mandatory-tag-base" title={t("This channel is mandatory")}>
-                  {t("mandatory")}
-                </span>
-              ) : null}
-              <ChannelAnchorLink id={c.id} newWindow={true} />
-            </div>
-          );
-        })}
+      {props.childChannelsId.map((channelId: number) => {
```
Comment (Etheryte) [inline web/html/src/manager/content-management/shared/components/panels/sources/channels/child-channels.tsx:27]: This code is unchanged, I simply removed iterating over the data twice.

**Review by ncounter — APPROVED** [review]
So, if I get it correctly, it is more or less a *scroll loading* implementation for slicing the whole bunch of data in pieces and incrementally add them to the view, once they are loaded the list will be just long as it is.
As an initial step it is already a good step ahead from my POV. Thanks for starting this journey :wink:

**Comment by Etheryte** [comment]
@ncounter Yes, the key difference here is that you can use the dropdown, type in it and search and get results, whereas previously the dropdown would also practically just freeze. Since the dropdown is searchable, I wouldn't really expect anyone to keep scrolling down for two days just to load all 5000 items. 😃 But if they did, it would still be at least somewhat manageable for the browser because it doesn't have to do all of the rendering at once.  

The virtualized scrolling bit I discussed under the ticket is planned for the rest of the view, but I'm only just starting trying to make that work so we'll see how that goes.

**Comment by Etheryte** [comment]
This PR is superseded by https://github.com/uyuni-project/uyuni/pull/4527.

**Pull request closed without merging**